dhis2/dhis2-core
# fix: broken app shell due to controller refactorings

### Summary

After the API controllers path was refactored to include the "/api" part, the string replacement of resource paths in the AppController would fail due to change in the input string now starting with "/api".
Another issue due to missing MimeMappings in the embedded Tomcat server came to light when the above issue was resolved. The embedded Tomcat's Mime mapping are now copied from the Spring Boot project.

## Automated tests
AppControllerTest#testGetInstalledAppIndexHtml

(There is extensive new code added to set up a test where we have a file system backed file store, instead of the default "transient" that is used in all other tests. This is in short accomplished by creating a ephemeral temporary directory on the test start up, where the app zip file get loaded into. This new app controller base test "AppControllerBaseTest", can be reused for other tests where we need to actually upload a real app zip file.)

## Manual test

1. Download the app .zip file from the Jira: [DHIS2-17365](https://dhis2.atlassian.net/browse/DHIS2-17365)
2. Navigate to the app manager and manually install the app
3. Observe that all bundled apps gets redirected through to the new app shell app.


[DHIS2-17365]: https://dhis2.atlassian.net/browse/DHIS2-17365?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
